### PR TITLE
🎉 New Source: Gitlab

### LINKED ISSUE #2305: Source: Gitlab
## Tell us about the new integration you’d like to have
I'd like to be able to pull data from the [Gitlab API](https://docs.gitlab.com/ee/api/README.html). The main streams I need to pull from are: 

* [Branches](https://docs.gitlab.com/ee/api/branches.html)
* [Commits](https://docs.gitlab.com/ee/api/commits.html)
* [Issues](https://docs.gitlab.com/ee/api/issues.html)
* [Pipelines](https://docs.gitlab.com/ee/api/pipelines.html)
* [Jobs](https://docs.gitlab.com/ee/api/jobs.html)
* [Projects](https://docs.gitlab.com/ee/api/projects.html)
* [Project milestones](https://docs.gitlab.com/ee/api/milestones.html)
* [Project Merge Requests](https://docs.gitlab.com/ee/api/merge_requests.html)
* [Users](https://docs.gitlab.com/ee/api/users.html)
* [Groups](https://docs.gitlab.com/ee/api/group_milestones.html)
* [Group Milestones](https://docs.gitlab.com/ee/api/users.html)
* [Group and Project members](https://docs.gitlab.com/ee/api/members.html)
* [Tags](https://docs.gitlab.com/ee/api/tags.html)
* [Releases](https://docs.gitlab.com/ee/api/releases/index.html)
* [Group Labels](https://docs.gitlab.com/ee/api/group_labels.html)
* [Project Labels](https://docs.gitlab.com/ee/api/labels.html)

There seems to exist a good Singer tap [here](https://gitlab.com/meltano/tap-gitlab)  that we can use. 

We'll need to create a Gitlab account and populate it with data to be able to test the API.

### PR DESCRIPTION
**Pull request by yevhenii-ldv**
## What
Created new source for pulling Gitlab streams [#2305](https://github.com/airbytehq/airbyte/issues/2305) via [Singer Tap](https://gitlab.com/meltano/tap-gitlab).

Supported streams:

* Branches
* Commits
* Issues
* Pipelines
* Jobs
* Projects
* Project milestones
* Project Merge Requests
* Users
* Groups
* Group Milestones
* Group and Project members
* Tags
* Releases
* Group Labels
* Project Labels
* Epics (only available for GitLab Ultimate and GitLab.com Gold accounts)
* Epic Issues (only available for GitLab Ultimate and GitLab.com Gold accounts)

## Pre-merge Checklist
- [ ] *Run integration tests*
- [ ] *Publish Docker images*

### CODE REVIEW

File: airbyte-integrations/connectors/source-gitlab-singer/source_gitlab_singer/source.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    tap_name = "Gitlab API"
+    api_error = HTTPError, Exception
+
+    def transform_config(self, raw_config):
+        return {**raw_config, "api_url": f'https://{raw_config["api_url"]}/'}
+
+    def try_connect(self, logger: AirbyteLogger, config: dict):
+        if not config["projects"] and not config["groups"]:
+            raise Exception("Either groups or projects need to be provided for connect to Gitlab API")
+        response = requests.get(f"{config['api_url']}/projects?private_token={config['private_token']}")
```
Comment (vitaliizazmic) [inline airbyte-integrations/connectors/source-gitlab-singer/source_gitlab_singer/source.py:46]: It is better to send "private_token" using the params keyword argument

**Comment by vitaliizazmic** [comment]
There are two streams only available for GitLab Ultimate and GitLab.com Gold accounts.  Does Tap properly handle this case? Do we have test cases for this types of account?

**Comment by yevhenii-ldv** [comment]
@vitaliizazmic Unfortunately, it is for these streams (Epics and Epic Issues) that there are no verification data for CI tests, since we have a regular Gitlab account.

**Comment by sherifnada** [comment]
Let's keep those streams for now. I'm fairly sure this tap is well maintained, so we can take the risk.

**Review by sherifnada — APPROVED** [review]
Nice! One comment then publish and merge when ready

File: airbyte-integrations/connectors/source-gitlab-singer/setup.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+from setuptools import find_packages, setup
+
+setup(
+    name="source_gitlab_singer",
+    description="Source implementation for Gitlab, built on the Singer tap implementation.",
+    author="Airbyte",
+    author_email="contact@airbyte.io",
+    packages=find_packages(),
+    install_requires=[
+        "tap-gitlab @ git+https://gitlab.com/meltano/tap-gitlab.git",
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-gitlab-singer/setup.py:34]: this should point to a specific version so we don't keep updating the tap under the hood

**Pull request merged**